app/cli/import_data.py

```diff
@@ -2152,6 +2154,60 @@ def fetch_missing_distributions(digest_path, out_dir, input_dir):
     L.info("%d distributions copied" % new_distributions)
 
 
+@cli.command()
+@click.argument("project_ids", type=REQUIRED_PATH)
+def assign_project(project_ids):
+    with Path(project_ids).open("r") as f:
+        project_ids = dict(line.strip().split(",") for line in f)
+
+    matching_projects = set()
+    with (
+        closing(configure_database_session_manager()) as database_session_manager,
+        database_session_manager.session() as db,
+    ):
+        rows = db.query(Entity).all()
+        for row in tqdm(rows):
+            matching_project = None
+            for project_id in project_ids.keys():
+                row_legacy_id = row.legacy_id
+                if row_legacy_id is None:
+                    continue
+                for row_legacy_id_elem in row_legacy_id:
+                    if project_id in row_legacy_id_elem:
+                        matching_project = project_id
+                        matching_projects.add(matching_project)
+                        break
+                if matching_project:
+                    row.authorized_public = False
+                    row.authorized_project_id = uuid.UUID(matching_project)
+                    db.flush()
+                    break
+        db.commit()
+        # MTypeClassification and ETypeClassification are public so nothing to do
+        # No activity imported so far
+
+        entities_with_asset = (
+            db.query(Entity)
+            .options(joinedload(Entity.assets))
```

Comment:
This `joinedload` causes an additional `left outer join asset` that seems unneeded, because there is already the explicit join with Asset below.

I think that it can be replaced with 
```suggestion
            .options(contains_eager(Entity.assets))
```

and it should also be possible to remove the `distinct()` in this way.